node/cmd/guardiand/adminserver.go

```diff
@@ -162,6 +162,43 @@ func tokenBridgeUpgradeContract(req *nodev1.BridgeUpgradeContract, timestamp tim
 	return v, nil
 }
 
+// wormchainStoreCode converts a nodev1.WormchainStoreCode to its canonical VAA representation
+// Returns an error if the data is invalid
+func wormchainStoreCode(req *nodev1.WormchainStoreCode, timestamp time.Time, guardianSetIndex uint32, nonce uint32, sequence uint64) (*vaa.VAA, error) {
+	// validate the length of the hex passed in
+	b, err := hex.DecodeString(req.WasmHash)
+	if err != nil {
+		return nil, errors.New("invalid cosmwasm bytecode hash (expected hex)")
```

Comment:
nit: we should wrap and return the actual error:

```suggestion
		return nil, fmt.Errorf("invalid cosmwasm bytecode hash (expected hex): %w", err)
```